File: dclib/core/src/main/java/dclib/util/IntRange.java
package dclib.util;

import com.badlogic.gdx.math.MathUtils;

public final class IntRange {

	private final int min;
	private final int max;

	public IntRange(final int min, final int max) {
		if (min > max) {
			throw new IllegalArgumentException("Min argument of " + min
					+ " cannot be greater than max argument of " + max);
		}
		this.min = min;
		this.max = max;
	}

	public final int min() {
		return min;
	}

	public final int max() {
		return max;
	}

	public final int difference() {
		return max - min;
	}

	public final float clamp(final int value) {
		return MathUtils.clamp(value, min, max);
	}

	public final int random() {
		return MathUtils.random(min, max);
	}

}

File: dclib/core/src/main/java/dclib/util/IntRange.kt
package dclib.util

import com.badlogic.gdx.math.MathUtils

class IntRange(val min: Int, val max: Int) {
    init {
        if (min > max) {
            throw IllegalArgumentException("Min argument of " + min
                    + " cannot be greater than max argument of " + max)
        }
    }

    operator fun contains(value: Int): Boolean {
        return value in min..max
    }

    fun difference(): Int {
        return max - min
    }

    fun clamp(value: Int): Float {
        return MathUtils.clamp(value, min, max).toFloat()
    }

    fun random(): Int {
        return MathUtils.random(min, max)
    }
}
